pkg/controller/postgresql.go

```diff
@@ -345,13 +349,18 @@ func (c *Controller) warnOnDeprecatedPostgreSQLSpecParameters(spec *acidv1.Postg
 	}
 
 	if len(spec.MaintenanceWindows) > 0 {
-		noeffect("maintenanceWindows", "Not implemented.")
+		noEffect("maintenanceWindows", "Not implemented.")
 	}
 
 	if (spec.UseLoadBalancer != nil || spec.ReplicaLoadBalancer != nil) &&
 		(spec.EnableReplicaLoadBalancer != nil || spec.EnableMasterLoadBalancer != nil) {
 		c.logger.Warnf("Both old and new load balancer parameters are present in the manifest, ignoring old ones")
 	}
+
+	isSmaller, err := util.RequestIsSmallerThanLimit(spec.ResourceRequests.Memory, spec.ResourceLimits.Memory)
+	if err == nil && !isSmaller {
+		notAllowed("memoryRequest", "higher than resource limit")
```

Comment:
That won't work because that will reject the legitimate case of request being equal to limit.

In `RequestIsSmallerThanLimit`, the line 
```go
 return request.Cmp(limit) == -1, nil
```
is responsible for that . `request.Cmp(limit)` will return zero if quantities are equal. Please check Kubernetes source code / docs for that method.